src/main/java/io/github/apace100/apoli/power/factory/condition/block/RedstoneInputCondition.java

```diff
@@ -0,0 +1,24 @@
+package io.github.apace100.apoli.power.factory.condition.block;
+
+import io.github.apace100.apoli.Apoli;
+import io.github.apace100.apoli.data.ApoliDataTypes;
+import io.github.apace100.apoli.power.factory.condition.ConditionFactory;
+import io.github.apace100.calio.data.SerializableData;
+import io.github.apace100.calio.data.SerializableDataTypes;
+import net.minecraft.block.pattern.CachedBlockPosition;
+import io.github.apace100.apoli.util.Comparison;
+
+public class RedstoneInputCondition {
+    public static boolean condition(SerializableData.Instance data, CachedBlockPosition block) {
+        return ((Comparison)data.get("comparison")).compare(block.getBlockEntity().getWorld().getReceivedRedstonePower(block.getBlockPos()), data.getFloat("compare_to"));
```

Comment:
You shouldn't be using `CachedBlockPosition#getBlockEntity` to get the world since it's nullable. This wouldn't work if the cached block doesn't have a block entity, and will cause a crash. You should use `CachedBlockPosition#getWorld` instead
```suggestion
        Comparison comparison = data.get("comparison");
        int compareTo = data.getInt("compare_to");

        int receivedRedstonePower = block.getWorld().getReceivedRedstonePower(block.getBlockPos());
        return comparison.compare(receivedRedstonePower, compareTo);
```